spec/lib/chat_review_queue_spec.rb

```diff
@@ -27,6 +27,56 @@
       }.to raise_error(Discourse::InvalidAccess)
     end
 
+    it "raises an error if the user already flagged the post" do
+      subject.flag_message(message, guardian, ReviewableScore.types[:spam])
+
+      second_flag_result =
+        subject.flag_message(message, guardian, ReviewableScore.types[:off_topic])
+
+      expect(second_flag_result[:success]).to eq(false)
+      expect(second_flag_result[:errors]).to contain_exactly(I18n.t("reviewables.already_handled"))
+    end
+
+    it "raises an error if a different user uses the same flag but we recently handled it" do
+      subject.flag_message(message, guardian, ReviewableScore.types[:spam])
+
+      reviewable = ReviewableChatMessage.last
+      reviewable.perform(admin, :ignore)
+
+      second_flag_result =
+        subject.flag_message(message, admin_guardian, ReviewableScore.types[:spam])
+
+      expect(second_flag_result[:success]).to eq(false)
+      expect(second_flag_result[:errors]).to contain_exactly(I18n.t("reviewables.already_handled"))
+    end
+
+    it "allow users to re-flag using the same flag type after the cooldown period" do
+      subject.flag_message(message, guardian, ReviewableScore.types[:spam])
+
+      reviewable = ReviewableChatMessage.last
```

Comment:
Might be worth some "update_reviable" helper

Reply:
Re-organizing tests make it better. I don't think we need a helper for the time being